lib/steady_state/attribute.rb

```diff
@@ -56,8 +56,14 @@ def steady_state(attr_name, predicates: true, states_getter: true, scopes: Stead
 
         delegate(*state_machines[attr_name].predicates, to: attr_name, allow_nil: true) if predicates
         if scopes
+          prefix = if scopes == true
+                     ''
+                   elsif scopes.is_a?(Hash) && scopes[:prefix]
+                     "#{scopes[:prefix] == true ? attr_name : scopes[:prefix]}_"
+                   end
```

Comment:
Sort of dense!  We are handling the following situations though:

```
scopes: true
scopes: false
scopes: { prefix: true }
scopes: { prefix: 'some_custom_prefix' }
```

Interestingly, `scopes: { prefix: false }` behaves like `scopes: true`.  Which I think is what I would expect.  

I also experimented a little with having a default `else` branch that raised if the key wasn't present.  For example, if you recently returned home from brunch and did `scopes: { prixefixe: true }` it just silently defaults to behavior equivalent to `scopes: true`.  If we wanted to be more supportive of users, a little defensive error there might be helpful such that we'd get:

```suggestion
          prefix = if scopes == true
                     ''
                   elsif scopes.is_a?(Hash) && scopes[:prefix]
                     "#{scopes[:prefix] == true ? attr_name : scopes[:prefix]}_"
                   else
                      raise ArgumentError, 'Invalid scopes option. Must be a Hash with a :prefix key or a boolean'
                   end
```

Reply:
On newer Rubies I tend to take nested hash options (like `:prefix`) and feed them directly into a new method as keyword arguments. This allows me to specify required/optional kwargs and get Ruby's built-in `ArgumentError` if an option is typoed.

E.g.:

```ruby
def build_prefix(prefix: false)
  prefix ? "#{prefix == true ? attr_name : prefix}_" : ""
end
```

And then the example above becomes:
```suggestion
          scopes = scopes.is_a?(Hash) ? scopes : {}
          prefix = build_prefix(**scopes)
```

Reply:
Oh, nice.  Will update!